File: library/src/main/java/me/proxer/library/api/DelimitedEnumSet.java
package me.proxer.library.api;

import com.squareup.moshi.JsonQualifier;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author Ruben Gees
 */
@Retention(RetentionPolicy.RUNTIME)
@JsonQualifier
public @interface DelimitedEnumSet {

    String delimiter() default " ";
}

File: library/src/main/kotlin/me/proxer/library/internal/adapter/DelimitedEnumSet.kt
package me.proxer.library.internal.adapter

import com.squareup.moshi.JsonQualifier

/**
 * @author Ruben Gees
 */
@JsonQualifier
@Retention(AnnotationRetention.RUNTIME)
internal annotation class DelimitedEnumSet(val delimiter: String = " ")
